src/test/compile-fail/borrowck/borrowck-overloaded-index-ref-index.rs

```diff
@@ -70,5 +70,5 @@ fn main() {
     };
     s[2] = 20;
     //[ast]~^ ERROR cannot assign to immutable indexed content
-    //[mir]~^^ ERROR cannot assign to immutable item
+    //[mir]~^^ ERROR cannot assign through immutable item
```

Comment:
I'm not sure about the look of this test though.

Reply:
This one seems ok-ish to me, but we might be able to improve upon it. I'll think about it.